prompt_poet/template_loaders.py

```diff
@@ -0,0 +1,74 @@
+"""A Prompt Poet (PP) template registry."""
+
+import os
+import jinja2 as j2
+
+
+class TemplateLoader:
+    """Base class for template loaders.
+
+    This abstract class defines the interface for loading templates from various sources.
+    Concrete implementations should handle specific template loading scenarios like
+    local filesystem, remote filesystem, etc.
+    """
+
+    def load(self) -> j2.Template:
+        """Load and return a Jinja2 template from the source."""
+        pass
+
+    def id(self) -> str:
+        """Generate a unique identifier for this template loader configuration."""
+        pass
+
+
+def _parse_template_path(template_path: str) -> tuple[str, str]:
+    """Parse the template path to determine the template source."""
+    template_dir, template_name = os.path.split(template_path)
+    if not template_dir:
+        template_dir = "."
+    return template_dir, template_name
+
+
+class LocalFSTemplateLoader(TemplateLoader):
+    """Template loader for loading templates from the local filesystem."""
+
+    def __init__(self, template_path: str):
+        self._template_dir, self._template_name = _parse_template_path(template_path)
+
+    def load(self) -> j2.Template:
+        try:
+            loader = j2.FileSystemLoader(searchpath=self._template_dir)
+        except j2.TemplateNotFound as ex:
+            raise j2.TemplateNotFound(
+                f"Template not found: {ex} {self._template_name=} {self._template_dir=}"
+            )
+
+        env = j2.Environment(loader=loader)
+        template = env.get_template(self._template_name)
+        return template
+
+    def id(self):
+        return f"{self._template_dir}/{self._template_name}"
+
+
+class LocalPackageTemplateLoader(TemplateLoader):
```

Comment:
i'm on the fence about whether we should have LocalPackageTemplateLoader at all...fundamentally it is just loading a template from local fs...and i could make a case that the extraction of the local fs path for package contents should be done upstream of the loader entirely. can you make the case for it?

Reply:
I actually think it is more organized to use `LocalPackageTemplateLoader` since there is a `j2.PackageLoader` just for that. extracting fs path from the package seems like an unnecessary step. wdyt

Reply:
Ok sg let’s go with it